tests/ttnn/unit_tests/operations/test_moreh_matmul.py

```diff
@@ -125,20 +142,91 @@ def moreh_matmul(params, has_output, compute_kernel_config, device):
         output=tt_output,
         compute_kernel_config=compute_kernel_config,
     )
-    tt_output_cpu = tt_output.cpu().to(cpu_layout).unpad_from_tile(output_shape).to_torch()
+    tt_output_cpu = ttnn.to_torch(tt_output)
 
     # torch matmul
     torch_out = torch.matmul(torch_input, torch_other)
 
     # test for equivalance
     rtol = atol = 0.1
-    passing, output_pcc = comp_allclose_and_pcc(torch_out, tt_output_cpu, pcc=0.999, rtol=rtol, atol=atol)
+    passing, output_pcc = comp_allclose_and_pcc(
+        torch_out, tt_output_cpu, pcc=0.999 if use_randint else 0.99, rtol=rtol, atol=atol
+    )
     logger.debug(f"Out passing={passing}")
     logger.debug(f"Output pcc={output_pcc}")
 
     return passing
 
 
+def moreh_matmul_backward(params, requires_grad, device, dtype=ttnn.bfloat16, use_randint=True):
+    input_shape, other_shape, output_shape = params
+    require_input_grad, require_other_grad = requires_grad
+
+    # get tensors
+    (
+        tt_input,
+        tt_other,
+        _,
+        tt_output_grad,
+        tt_input_grad,
+        tt_other_grad,
+        torch_input,
+        torch_other,
+        torch_output_grad,
+    ) = get_tensors(
+        input_shape,
+        other_shape,
+        output_shape,
+        require_input_grad,
+        require_other_grad,
+        False,
+        device,
+        use_randint=use_randint,
+        npu_dtype=dtype,
+    )
+
+    # torch matmul
+    torch_out = torch.matmul(
+        torch_input.requires_grad_(require_input_grad), torch_other.requires_grad_(require_other_grad)
+    )
+    torch_out.backward(torch_output_grad)
+
+    # tt matmul backward
+    tt_input_grad, tt_other_grad = ttnn.operations.moreh.matmul_backward(
+        tt_output_grad,
+        tt_input,
+        tt_other,
+        are_required_outputs=(require_input_grad, require_other_grad),
+        input_a_grad=tt_input_grad,
+        input_b_grad=tt_other_grad,
+    )
+
+    # test for equivalance
+    rtol = atol = 0.1
+    cpu_layout = ttnn.ROW_MAJOR_LAYOUT
+    if require_input_grad:
+        ttcpu_input_grad = tt_input_grad.cpu().to(cpu_layout).unpad_from_tile(input_shape).to_torch()
```

Comment:
Can you use `ttnn.to_torch(tt_input_grad,...)` here?

Reply:
Done